cmd/flakechecker/flake_config.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"io"
+	"os"
+	"regexp"
+)
+
+// flakeCheckerRecord represents configuration record used by flakechecker to evaluate failed tests
+//
+// It contains the following fields:
+// match_job_name - name of the job that should be evaluated by flakechecker. i.e. (branch should be evaluated, but main not)
+// ratio_job_name - job name that should be used for ratio calculation. i.e. we take main branch test runs as base for evaluation of flake ratio
+// test_name_regex - regex used to match test names. Some test names contain detailed information (i.e. version 4.4.4), but we want to use ratio for all tests in that group (i.e. 4.4.z). Using regex allow us to group tests differently.
+// classname - class name of the test that should be isolated. With this option we can isolate single flake test from suite and isolate only that one from the rest.
+// ratio_threshold - failure percentage that is allowed for this test. This information is usually fetched from historical executions and data collected in DB.
+//
+// This record also contains helper fields where we keep compiled regex.
+type flakeCheckerRecord struct {
+	MatchJobName   string `json:"match_job_name"`
+	RatioJobName   string `json:"ratio_job_name"`
+	TestNameRegex  string `json:"test_name_regex"`
+	Classname      string `json:"classname"`
+	RatioThreshold int    `json:"ratio_threshold"`
+
+	regexMatchJobName  *regexp.Regexp
+	regexTestNameRegex *regexp.Regexp
+}
+
+func (r *flakeCheckerRecord) updateRegex() error {
+	validRegex, err := regexp.Compile(fmt.Sprintf("^%s$", r.MatchJobName))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("invalid flake config match job regex: %v", r))
+	}
+	r.regexMatchJobName = validRegex
+
+	validRegex, err = regexp.Compile(fmt.Sprintf("^%s$", r.TestNameRegex))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("invalid flake config test name regex: %v", r))
+	}
+	r.regexTestNameRegex = validRegex
+
+	return nil
+}
+
+func (r *flakeCheckerRecord) matchJobName(jobName string) (bool, error) {
+	if r.regexMatchJobName == nil {
```

Comment:
can we get rid of this is we wrap `updateRegex body` with once?

Reply:
I think we don't need this as in `loadFlakeConfigFile` we update regex so it should not be nil

Reply:
The primary reason for having this check was to simplify testing. I also didn't like it.

I have changed it and separated the configuration from JSON and the structure used as an object. It should be cleaner now.